functions/src/index.ts

```diff
@@ -37,6 +37,27 @@ function matchInArray(regex: RegExp, arrayOfString: string[]): string {
   return '';
 }
 
+export const onTagDeleted = functions.firestore.document('courses/{courseID}').onUpdate(async (snapshot, context) => {
+  const prev_data = snapshot.before.data();
+  const new_data = snapshot.after.data();
+  const prev_tags = prev_data ? prev_data.tags : [];
+  const new_tags = new_data ? new_data.tags : [];
+  for (const tag of prev_tags){
+    if (!new_tags.includes(tag)){
+      admin.firestore().collection('questions').where('course', '==', parseInt(context.params.courseID)).get()
```

Comment:
Please rewrite this part using awaits (look in other functions). The last promise should be the return value for the function, otherwise Firebase will kill when the functions exits, even if the update is not done

Reply:
isn't it better to return Promise.all of all promises?

Reply:
Yes it is :)